lua/script.lua

```diff
@@ -121,5 +84,52 @@ Script.metadata = function(filename)
   return meta
 end
 
+--- signal error in script load
+function Script.error(msg)
+  Script.errorstate = true
+  Script.errormsg = msg
+end
+
+-- read system.lua and set parameters back to stored vals
+function Script.resume()
+  local f = io.open(data_dir..'system.lua')
+  if f ~= nil then
+    io.close(f)
+    dofile(data_dir..'system.lua')
+  end
+
+  if Script.file== nil or Script.file== '' then
```

Comment:
(nit) add some spaces:

    if Script.file == nil or Script.file == '' then

also, we should really have a `string.isEmpty(..)` function somewhere?